plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/search/JavaSearchJsonRpcClient.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ext.java.client.search;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.eclipse.che.api.core.jsonrpc.commons.RequestTransmitter;
+import org.eclipse.che.api.promises.client.Promise;
+import org.eclipse.che.ide.ext.java.shared.dto.search.FindUsagesRequest;
+import org.eclipse.che.ide.ext.java.shared.dto.search.FindUsagesResponse;
+import org.eclipse.che.ide.ui.loaders.request.LoaderFactory;
+import org.eclipse.che.ide.ui.loaders.request.MessageLoader;
+
+import static org.eclipse.che.api.promises.client.callback.AsyncPromiseHelper.createFromAsyncRequest;
+
+/**
+ * WebSocket implementation of JavaSearchService
+ */
+@Singleton
+public class JavaSearchJsonRpcClient implements JavaSearchService {
+
+    private static final int    TIMEOUT               = 20_000;
+    private static final String ENDPOINT_ID           = "ws-agent";
+    private static final String METHOD_NAME           = "javaSearch/findUsages";
```

Comment:
I think will be better if this method name will be declared one time, for example you can move it to `org.eclipse.che.ide.ext.java.shared.Constants` and use it for the server side and for the client.